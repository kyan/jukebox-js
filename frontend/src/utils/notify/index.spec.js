import { store } from 'react-notifications-component'
import NotifyMe from './index'

describe('notify', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('calls the success toast interface', () => {
    jest.spyOn(store, 'addNotification')
    NotifyMe.success({ id: '123', title: 'Voting', message: 'this is a vote' })
    expect(store.addNotification.mock.calls[0]).toMatchSnapshot()
  })

  it('calls the info toast interface', () => {
    jest.spyOn(store, 'addNotification')
    NotifyMe.info({ id: '123', title: 'Voting', message: 'this is a vote' })
    expect(store.addNotification.mock.calls[0]).toMatchSnapshot()
  })

  it('calls the warning toast interface', () => {
    jest.spyOn(store, 'addNotification')
    NotifyMe.warning({ id: '123', title: 'Voting', message: 'this is a vote' })
    expect(store.addNotification.mock.calls[0]).toMatchSnapshot()
  })
})
