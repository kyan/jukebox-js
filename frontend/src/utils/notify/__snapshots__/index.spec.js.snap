// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notify calls the info toast interface 1`] = `
Array [
  Object {
    "animationIn": Array [],
    "animationOut": Array [],
    "container": "bottom-left",
    "dismiss": Object {
      "click": true,
      "duration": 3000,
      "onScreen": false,
      "showIcon": false,
      "touch": true,
      "waitForAnimation": false,
    },
    "id": "123",
    "insert": "top",
    "message": "this is a vote",
    "onRemoval": [Function],
    "slidingEnter": Object {
      "delay": 0,
      "duration": 600,
      "timingFunction": "linear",
    },
    "slidingExit": Object {
      "delay": 0,
      "duration": 600,
      "timingFunction": "linear",
    },
    "title": "Voting",
    "touchRevert": Object {
      "delay": 0,
      "duration": 600,
      "timingFunction": "linear",
    },
    "touchSlidingExit": Object {
      "fade": Object {
        "delay": 0,
        "duration": 300,
        "timingFunction": "linear",
      },
      "swipe": Object {
        "delay": 0,
        "duration": 600,
        "timingFunction": "linear",
      },
    },
    "type": "info",
  },
]
`;

exports[`notify calls the success toast interface 1`] = `
Array [
  Object {
    "animationIn": Array [],
    "animationOut": Array [],
    "container": "bottom-left",
    "dismiss": Object {
      "click": true,
      "duration": 3000,
      "onScreen": false,
      "showIcon": false,
      "touch": true,
      "waitForAnimation": false,
    },
    "id": "123",
    "insert": "top",
    "message": "this is a vote",
    "onRemoval": [Function],
    "slidingEnter": Object {
      "delay": 0,
      "duration": 600,
      "timingFunction": "linear",
    },
    "slidingExit": Object {
      "delay": 0,
      "duration": 600,
      "timingFunction": "linear",
    },
    "title": "Voting",
    "touchRevert": Object {
      "delay": 0,
      "duration": 600,
      "timingFunction": "linear",
    },
    "touchSlidingExit": Object {
      "fade": Object {
        "delay": 0,
        "duration": 300,
        "timingFunction": "linear",
      },
      "swipe": Object {
        "delay": 0,
        "duration": 600,
        "timingFunction": "linear",
      },
    },
    "type": "success",
  },
]
`;

exports[`notify calls the warning toast interface 1`] = `
Array [
  Object {
    "animationIn": Array [],
    "animationOut": Array [],
    "container": "bottom-left",
    "dismiss": Object {
      "click": true,
      "duration": 5000,
      "onScreen": false,
      "showIcon": false,
      "touch": true,
      "waitForAnimation": false,
    },
    "id": "123",
    "insert": "top",
    "message": "this is a vote",
    "onRemoval": [Function],
    "slidingEnter": Object {
      "delay": 0,
      "duration": 600,
      "timingFunction": "linear",
    },
    "slidingExit": Object {
      "delay": 0,
      "duration": 600,
      "timingFunction": "linear",
    },
    "title": "Voting",
    "touchRevert": Object {
      "delay": 0,
      "duration": 600,
      "timingFunction": "linear",
    },
    "touchSlidingExit": Object {
      "fade": Object {
        "delay": 0,
        "duration": 300,
        "timingFunction": "linear",
      },
      "swipe": Object {
        "delay": 0,
        "duration": 600,
        "timingFunction": "linear",
      },
    },
    "type": "warning",
  },
]
`;
